pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -329,6 +329,22 @@ type ClusterCephxConfig struct {
 	// Daemon configures CephX key settings for local Ceph daemons managed by Rook and part of the
 	// Ceph cluster. Daemon CephX keys can be rotated without affecting client connections.
 	Daemon CephxConfig `json:"daemon,omitempty"`
+
+	// CSI configures CephX key rotation settings for the Ceph-CSI daemons in the current Kubernetes cluster.
+	// CSI key rotation can affect existing PV connections, so take care when exercising this option.
+	CSI CephXConfigWithPriorCount `json:"csi,omitempty"`
+}
+
+type CephXConfigWithPriorCount struct {
+	CephxConfig `json:",inline"` // inline core CephxConfig
+
+	// PriorKeyCount tells Rook how many prior keys to keep active.
+	// Generally, this would be set to 1 to allow for a migration period for applications.
+	// If desired, set this to 0 to delete prior keys after migration
```

Comment:
```suggestion
	// PriorKeyCount tells Rook how many prior keys to keep active.
	// Generally, this would be set to 1 to allow for a migration period for applications.
	// If desired, set this to 0 to delete prior keys after migration.
	// This config only applies to prior keys that already exist. If PriorKeyCount is set to 2 while only a single key currently exists, only a single prior key will be kept, and the reported status will only indicate the actual number of prior keys, not necessarily a reflection of PriorKeyCount config here. 
```

This is probably an important point to make for users so they don't worry if input and status differ. 

I wonder if we should change the config name to `keepPriorKeyCountMax` or `priorKeyCountMax` to reflect this.